lib/models/purchase_result.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:equatable/equatable.dart';
+
+import 'customer_info_wrapper.dart';
+import 'store_transaction.dart';
+
+/// Represents the successful result of a purchase operation.
+class PurchaseResult extends Equatable {
+  /// The updated [CustomerInfo] after the purchase has been synced with
+  /// RevenueCat's servers.
+  final CustomerInfo customerInfo;
+  /// The [StoreTransaction] for this purchase.
+  final StoreTransaction storeTransaction;
```

Comment:
I see that in the web platforms (RN and capacitor) we also have the `productIdentifier` here... but seemed redundant since it's already part of the `PurchaseResult`.